src/DotVVM.Framework/Routing/DotvvmRouteTable.cs

```diff
@@ -65,14 +65,17 @@ public DotvvmRouteTable GetGroup(string groupName)
         /// <param name="content">Contains routes to be added</param>
         public void AddGroup(string groupName, string urlPrefix, string virtualPathPrefix, Action<DotvvmRouteTable> content)
         {
-            if (groupName == null || groupName == "")
+            if (string.IsNullOrEmpty(groupName))
+            {
                 throw new ArgumentNullException("Name of the group cannot be null or empty!");
+            }
             if (routeTableGroups.ContainsKey(groupName))
             {
                 throw new InvalidOperationException($"The group with name '{groupName}' has already been registered!");
             }
-            urlPrefix = group?.UrlPrefix + ((urlPrefix == null || urlPrefix == "") ? "" : urlPrefix + "/");
-            virtualPathPrefix = group?.VirtualPathPrefix + ((virtualPathPrefix == null || virtualPathPrefix == "") ? "" : virtualPathPrefix + "/");
+            urlPrefix = CombinePath(group?.UrlPrefix, urlPrefix);
```

Comment:
Note that this is technically a runtime breaking change - it does not append the slash to UrlPrefix anymore on initialization.

Reply:
Url which ends with a slash is forbidden and exception will be thrown (if I understand your note)

Reply:
I mean the format of data in `UrlPrefix` has changed.

Reply:
Okay, I get it